src/Sylius/Bundle/PaymentBundle/Provider/AbstractServiceProvider.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PaymentBundle\Provider;
+
+use Sylius\Bundle\PaymentBundle\Exception\PaymentRequestNotSupportedException;
+use Sylius\Bundle\ResourceBundle\Controller\RequestConfiguration;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+/** @experimental */
+abstract class AbstractServiceProvider implements ServiceProviderAwareProviderInterface
```

Comment:
We need to think about simplifying these, and maybe split the logic into a few different services, as it is it's pretty magical and it's easy to get lost

Reply:
We will have to rethink this service and its implementation and probably refactor it, but after the merge of this branch into 2.0 🖖🏻